tiny/lisp.c

```diff
@@ -501,7 +501,7 @@ void print(FILE *f, value_t v)
     value_t cd;
 
     switch (tag(v)) {
-    case TAG_NUM: fprintf(f, "%ld", numval(v)); break;
+    case TAG_NUM: fprintf(f, "%lld", numval(v)); break;
```

Comment:
Compiling with both clang and gcc on macOS 10.12.6.
Not experiencing this warning.

I do however see warnings only with clang, both the system one and version 4.0.1.
The warning is about the usage of the macro fixnum:
```log
In file included from flisp.c:748:
./equal.c:146:20: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
            return fixnum(-1);
                   ^~~~~~~~~~
./flisp.h:55:45: note: expanded from macro 'fixnum'
#define fixnum(x) ((value_t)(((fixnum_t)(x))<<2))
                             ~~~~~~~~~~~~~~~^
```
That warning appears 16 times, only when compiling with clang.
Gcc does not even move an eye-lid on the matter.